### ISSUE TITLE
Bug with clean.list

### ISSUE THREAD

**Issue by funilrys** [dead-hosts/cameleon_web_sysctl.org#2]
@dnmTx said (https://github.com/Ultimate-Hosts-Blacklist/dev-center/issues/6#issue-347250763):

> @mitchellkrogza there is a issue with the formatting in the output folder.I'm using the lists with
> (g)awk to to filter the duplicates against StevenBlack's lists which are a lot ( like 13 000+) and simply does not work.It outputs the same number as the input,different command combinations results in wrong numbers.If i download the lists from the clean.list works like charm.I'm using (g)awk with other lists as well and have no problems but here,and only if i fetch it from the output folder. @funilrys's clean.list is empty so i had to start downloading it from here but his repo got the same formatting issue in the output folder as well.
> If for some reason you don't believe me or can't do nothing about it.please,at least keep the clean.list populated and most importantly updated so i can be able to download it and filter it.Thank you.

**Comment by funilrys** [comment]
@dnmTX May I ask what do you mean with the following ?

> different command combinations results in wrong numbers

**Comment by funilrys** [comment]
Thanks for reporting @dnmTX, it is an issue which is present into this infrastructure only. The issue has something to do with the changed behind our cross repository system.

Indeed, I develop both Ultimate and Dead-Hosts organization separately. That ensure that I do not produce a world wide issue as Ultimate is growing much faster than Dead-Hosts.

I found and can confirm the issue. I will address the fix before the end of this day (Berlin Time) in all our infrastructure which means that everything will work correctly in the coming days after the fix has been pushed. 

Sorry for the pain it produces, I will implement the necessary control system in my backend and system check so that It will never happened again.

Sorry for the disagreement.

Cheers,
Nissar.

P.S: Stay in touch, I will close this issue only once the fix will be pushed.

**Comment by dnmTX** [comment]
@funilrys  this is the  command that i'm using and it's proven working.Works with `clean.list` and several others that needs filtering before use on my end:
`awk 'NR==FNR{a[$0];next}!($0 in a)'` stevenblack.txt cameleon.txt > no-duplicates-in-cameleon.txt

The closest i've tried which gets very close is:
`awk 'NR==FNR{a[$2];next} ($2 in a)'`  it matches the second column instead of the whole line but it produces more lines then suppose to aka some duplicate entries got trough i'm guessing. So....
....the fact that the first mentioned command works everywhere BUT in lists from the `output` folder tells me that the problem is somewhere in that folder not with the command itself.

**Comment by dnmTX** [comment]
@funilrys which issue??? The empty `clean.list` or the formatting?

**Comment by funilrys** [comment]
@dnmTX If I understood correctly the formatting issue is because you are looking for a file in hosts format ? 
Normally we also generate a list of element for each status. So if I fix the clean.list issue, I probably also fix your formatting issue. Or am I not understanding your explanation ?

**Comment by dnmTX** [comment]
@funilrys  `clean.list` is the only one that works for me.The issue with it was that in dead-hosts was empty,there is nothing wrong with the formatting in it because **awk** has no problems filtering it.I'm just hoping that whatever changes you going to make it will not affect the `clean.list` as it is my only source as of now.

**Comment by funilrys** [comment]
@dnmTX  As tested at https://github.com/dead-hosts/repository-structure my fix regenerate the `clean.list` correctly. 

I'm currently resynchronizing all repositories. The generation for each sources will depends on their size but at the end the `clean.list` will be generated for good. 
You can follow the building process at https://travis-ci.org/dead-hosts.

Again, sorry for the disagreement my last repository synchronization and my review of the cross repository configuration system caused.

Cheers,
Nissar

P.S: The main issue (for me) is now fixed but will wait until the repository you mentioned generate a full `clean.list` before closing this. About the formatting issue I would need to know which files you sourced to have formatting issues. Feel free to document that here or at https://github.com/funilrys/PyFunceble as PyFunceble is the only one responsible if there is an issue into the `output` directory.

**Comment by dnmTX** [comment]
@funilrys  i tested them all ACTIVE ones:
https://github.com/Ultimate-Hosts-Blacklist/cameleon_at_sysctl.org/blob/master/output/domains/ACTIVE/list

https://github.com/Ultimate-Hosts-Blacklist/cameleon_at_sysctl.org/blob/master/output/hosts/ACTIVE/hosts

**Comment by funilrys** [comment]
@dnmTX Thanks will investigate further. I assume that `stevenblack.txt` is the unified hosts file from Steve. Is that true ?

**Comment by dnmTX** [comment]
@funilrys  yes,and this was just example so to be more self explanatory.I don't really use the same names when i download the lists.This is the link where i'm getting the lists from and have no problems with it filtering it against other lists:
https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/fakenews/hosts

**Comment by dnmTX** [comment]
@funilrys  thank you for fixing it.Let me know if you find anything on the formatting end.

**Comment by dnmTX** [comment]
@funilrys  looks like the `clean.list` is not updating again on both repos. Can you check please.

**Comment by funilrys** [comment]
Hi @dnmTX due to a Travis CI issue (not related to my code but to the container), the build stopped. 

I manually restarted them. The clean.list should be generated asap.
I did not had time this week for me, so I wasn't unable to manually check the Travis CI infrastructure.
